src/main/scala/com/datastax/sparkstress/ReadTask.scala

```diff
@@ -39,27 +46,124 @@ abstract class ReadTask(config: Config, sc: SparkContext) extends StressTask {
   val coresPerNode:Int = defaultParallelism / numberNodes
   def run()
 
-  def runTrials(sc: SparkContext): Seq[TestResult] = {
+  def runTrials(ss: SparkSession): Seq[TestResult] = {
     println("About to Start Trials")
     for (trial <- 1 to config.trials) yield {
       TestResult(time(run()),0L)
     }
   }
 }
 
+/**
+  * Supertype used to encapsulate reading logic for all dataset read methods.
+  */
+abstract class DatasetReadTask(config: Config, ss: SparkSession) extends ReadTask(config, ss) {
+  import org.apache.spark.sql.functions._ // needed to use col()
+  def read_columns(columnNames: Seq[String]): Long = {
+    val columns: Seq[org.apache.spark.sql.Column] = columnNames.map(col(_))
+    config.saveMethod match {
```

Comment:
why does save method change read method?

Reply:
Yeah I guess I could generalize with 
```spark.read.format("<save_method>").load(s"dsefs:///${keyspace}.${table}").select(columns:_*)count``` for the DSEFS cases